homeassistant/components/refoss/coordinator.py

```diff
@@ -0,0 +1,274 @@
+                payload={},
+            )
+
+            abilities = res.get("payload", {}).get("ability")
+
+        except Exception as e:
+            LOGGER.warning(
+                f"Device %s (%s) is online, but timeout occurred "
+                f"when fetching its abilities. Reason: %s",
+                str(device_info.dev_name),
+                str(device_info.uuid),
+                e,
+            )
+        if abilities is not None:
+            device = build_meross_device_from_abilities(
+                http_device_info=device_info, device_abilities=abilities
+            )
+
+        if device is not None:
+            self.enroll_device(device)
+
+        return device
+
+    def enroll_device(self, device: BaseDevice):
+        if device.uuid in self._devices_by_internal_id:
+            return
+        else:
+            self._devices_by_internal_id[device.uuid] = device
+
+    def lookup_base_by_uuid(self, device_uuid: str) -> None:
+        res = list(
+            filter(
+                lambda d: d.uuid == device_uuid,
+                self._devices_by_internal_id.values(),
+            )
+        )
+
+        if len(res) > 1:
+            LOGGER.warning(f"Multiple devices found for device_uuid {device_uuid}")
+            return None
+        elif len(res) == 1:
+            return res[0]
+        else:
+            return None
+
+    async def async_device_discovery(
+        self,
+        meross_device_uuid: str = None,
+        cached_http_device_list: List[HttpDeviceInfo] = None,
+    ) -> List[BaseDevice]:
+        if cached_http_device_list is None:
+            http_devices = self.merossSocket.async_socket_find_devices()
+        else:
+            http_devices = cached_http_device_list
+
+        if meross_device_uuid is not None:
+            http_devices = filter(lambda d: d.uuid == meross_device_uuid, http_devices)
+
+        res = []
+        for device in http_devices:
+            if self.lookup_base_by_uuid(device.uuid) is not None:
+                res.append(self.lookup_base_by_uuid(device.uuid))
+                continue
+
+            dev = await self._async_enroll_new_http_dev(device)
+
+            if dev is not None:
+                res.append(dev)
+
+        return res
+
+    def HandlePushState(self, event: threading.Event):
+        while True:
+            if event.is_set():
+                break
+            if len(pushStateDataList) == 0:
+                continue
+
+            data = pushStateDataList.pop(0)
+
+            if data is not None:
+                try:
+                    namespace = data.get("header", {}).get("namespace")
+                    uuid = data.get("header", {}).get("uuid")
+                    method = data.get("header", {}).get("method")
+                    payload = data.get("payload")
+                    if namespace is None or uuid is None or payload is None:
+                        continue
+
+                    if method != "PUSH":
```

Comment:
in .const
```.const
PUSH = "PUSH"
```
and then this line:
```suggestion
                    if method != PUSH:
```